src/upload.cpp

```diff
@@ -110,6 +115,180 @@ void Upload::onTeleTANFinished()
     reply->deleteLater();
 }
 
+void Upload::uploadGUID(QString const &guid)
+{
+    if (((m_status == StatusIdle) || (m_status == StatusError)) && (m_reply == nullptr)) {
+        bool valid = validateGUID(guid);
+        if (valid) {
+            setStatus(StatusSubmitGUID);
+            submitGUID(guid);
+        }
+        else {
+            qDebug() << "Invalid GUID";
+            setStatusError(ErrorInvalidGUID);
+        }
+    }
+    else {
+        // Do nothing until the existing upload completes
+        qDebug() << "Upload already in progress";
+    }
+}
+
+void Upload::submitGUID(QString const &guid)
+{
+    QCryptographicHash hasher(QCryptographicHash::Sha256);
+    hasher.addData(guid.toLatin1());
+    QString hashedGuid = static_cast<QString>(hasher.result());
```

Comment:
The server expects the guid to be uppercase hexadecimal and to include the "-" characters, so you'll need to add them in here before you perform the hash.